app/jobs/irs_attempts_events_batch_job.rb

```diff
@@ -0,0 +1,30 @@
+class IrsAttemptsEventsBatchJob < ApplicationJob
+  queue_as :default
+
+  def perform(timestamp: Time.zone.now - 1.hour, dir_path: './attempts_api_output')
+    return nil unless IdentityConfig.store.irs_attempt_api_enabled
+
+    events = IrsAttemptsApi::RedisClient.new.read_events(timestamp: timestamp)
+    event_values = events.values.join("\r\n")
+
+    decoded_key_der = Base64.strict_decode64(IdentityConfig.store.irs_attempt_api_public_key)
+    pub_key = OpenSSL::PKey::RSA.new(decoded_key_der)
+
+    result = IrsAttemptsApi::EnvelopeEncryptor.encrypt(
+      data: event_values, timestamp: timestamp, public_key: pub_key,
+    )
+
+    # write to a file and store on the disk until S3 is setup
+    FileUtils.mkdir_p(dir_path)
+
```

Comment:
in hindsight, this `mkdir_p` was masking the bug with the tmpdir being cleaned up beforehand, maybe we should remove it. that prevents the job from writing files where it's not supposed to 

```suggestion

```

Reply:
Do we actually need to get rid of this? I can see an argument for doing so, that whatever calls this shouldn't pass in a directory that doesn't exist. Is there some other way we can ensure safety / error handling if we remove this line?

Reply:
I think that this code will evolve over time. Eventually we'll write a temp file and upload that to S3. So in that case, the code could make it's own tmpdir and clean that up.

However in the meantime, this code hid a bug so I'd vote to remove it.